src/main.rs

```diff
@@ -130,16 +187,44 @@ fn optimize(matches: clap::ArgMatches) {
         }
     };
 
-    let rules = rules_from_str(split_rules);
+    // Get multi-pattern rules. learned_rules are the learned rules from TASO, pre_defined_multi are the hand-specified rules from TASO
+    let multi_patterns = match matches.value_of("multi_rules") {
+        Some(rule_file) => {
+            let learned_rules =
+                read_to_string(rule_file).expect("Something went wrong reading the rule file");
+            let pre_defined_multi = PRE_DEFINED_MULTI.iter().map(|&x| x);
+            let multi_rules: Vec<&str> =
+                learned_rules.split("\n").chain(pre_defined_multi).collect();
+            MultiPatterns::with_rules(multi_rules)
+        }
+        None => {
+            let multi_rules: Vec<&str> = PRE_DEFINED_MULTI.iter().map(|&x| x).collect();
+            MultiPatterns::with_rules(multi_rules)
+        }
+    };
```

Comment:
When you have a match with only two branches, one of them not binding any variables, consider using `if .. let ... else ...` instead of match. It helps readability, especially when one the `else` branch is the *default* path (here the `None`).

```
let multi_patterns = if let Some(rule_file) = matches.value_of("multi_rules") {
  ...
} else {
  ...
}
```

Reply:
Done